File: 2581-divide-players-into-teams-of-equal-skill/2581-divide-players-into-teams-of-equal-skill.java
class Solution {
    public long dividePlayers(int[] skill) {
        Arrays.sort(skill);
        int n=skill.length,sum=skill[0]+skill[n-1];
        long ans=skill[0]*skill[n-1];
        for(int i=1;i<n/2;i++){
            if(sum==skill[i]+skill[n-1-i]) ans+=(skill[i]*skill[n-1-i]);
            else return -1;
        }
        return ans;
    }
}

File: 2581-divide-players-into-teams-of-equal-skill/2581-divide-players-into-teams-of-equal-skill.cpp
class Solution {
public:
    long long dividePlayers(vector<int>& skill) {
        sort(skill.begin(),skill.end());
        int n=skill.size(),sum=skill[0]+skill[n-1];
        long long ans=skill[0]*skill[n-1];
        for(int i=1;i<n/2;i++){
            if(sum==skill[i]+skill[n-1-i]) ans+=skill[i]*skill[n-1-i];
            else return -1;
        }
        return ans;
    }
};